src/Symfony/Component/Security/Core/Authentication/Provider/DaoAuthenticationProvider.php

```diff
@@ -56,7 +57,7 @@ protected function checkAuthentication(UserInterface $user, UsernamePasswordToke
         $currentUser = $token->getUser();
         if ($currentUser instanceof UserInterface) {
             if ($currentUser->getPassword() !== $user->getPassword()) {
-                throw new BadCredentialsException('The credentials were changed from another session.');
+                throw new AccountPasswordChangedException('The credentials were changed from another session.');
```

Comment:
What if we remove the token on the BadCredentialsException as well? This change will be done on [ExceptionListener.php L197](https://github.com/symfony/symfony/blob/1ab5be70ae0128c9603401086e5a4727548ccc95/src/Symfony/Component/Security/Http/Firewall/ExceptionListener.php#L197) ??